### ISSUE TITLE
authorization fails when `cargo publish` is used

### ISSUE THREAD

**Issue by JHenneberg** [go-gitea/gitea#27333]
### Description

I want to publish a crate to the gitea package registry 
`cargo publish --registry gitea` which returns

```
failed to get a 200 OK response, got 401
  headers:
        HTTP/1.1 401 Unauthorized
        Content-Type: text/plain; charset=utf-8
        X-Content-Type-Options: nosniff
        Date: Thu, 28 Sep 2023 10:23:41 GMT
        Content-Length: 17

  body:
  authGroup.Verify
```

A token is used with complete access and it is defined in `credentials.toml`.

The repository  `_cargo_index` is existing and has `config.json` with `dl` and `api` as entries

What kept me wondering is when I do not have a credentials.toml for a token is asked.
And I should go `http://{url}/api/packages/{USER}/cargo/me` which returns always 404. 
This process of following an url to get access to a token is not explained in the docs.

Thats why i created my own token with full access without success.

### Gitea Version

1.20.4

### Can you reproduce the bug on the Gitea demo site?

Yes

### Git Version

2.40.1

### Operating System

Win 11

### How are you running Gitea?

Running in Docker

### Database

SQLite

**Comment by KN4CK3R** [comment]
Does the `config.json` in the index repository contain a `auth-required: true` field? If not you need to recreate that file from Settings > Packages > Initialize Index. The visibility change of an owner is sadly not synced into the config.

To avoid this, you can use the sparse http index for cargo which does not use the index repository.

**Comment by JHenneberg** [comment]
After Initializing  the `config.json` is still not containing `auth-required: true`.
The owner of the repositroy is `Owner` to which I belong. 
I have 2FA activated so I need to use anyway all the time a token. 
I am a bit confused. There are two different authorizations? The usual one to get access to Gitea and then a different one publishing a crate or why there need to be `auth-required: true`? I thought I just need to have access to the repository and then by using `cargo publish --registry gitea` this will make the corresponding commit to the index.

my config.toml:
```toml
[registry]
default = "gitea"

[registries.gitea]
index = "http://<URL>/<ORGANIZATION>/_cargo-index.git"

[net]
git-fetch-with-cli = true
```

**Comment by JHenneberg** [comment]
I created an extra folder `.cargo` in the root of the repositry and placed `config.toml` inside, which is fine (acc. to https://doc.rust-lang.org/cargo/reference/registries.html but not acc. to https://docs.gitea.com/usage/packages/cargo), but I also placed the `.credentials.toml` inside and published with `--allow-dirty` because the credential file was of course not checked in. Instead the `.credentials.toml` should be place in the home directory as stated in both documentations.
